cpp/src/arrow/io/buffered.h

```diff
@@ -41,12 +41,12 @@ class ARROW_EXPORT BufferedOutputStream : public OutputStream {
 
   /// \brief Create a buffered output stream wrapping the given output stream.
   /// \param[in] raw another OutputStream
-  /// \param[in] buffer_size the size of the temporary buffer. Allocates from
-  /// the default memory pool
+  /// \param[in] buffer_size the size of the temporary write buffer
+  /// \param[in] pool a MemoryPool to use for allocations
   /// \param[out] out the created BufferedOutputStream
   /// \return Status
   static Status Create(std::shared_ptr<OutputStream> raw, int64_t buffer_size,
-                       std::shared_ptr<BufferedOutputStream>* out);
+                       MemoryPool* pool, std::shared_ptr<BufferedOutputStream>* out);
```

Comment:
I think the most used convention in our codebase is to pass the MemoryPool in the first parameter.

Reply:
Since both `raw` and `pool` are in-out parameters, they should appear after `buffer_size` (in parameter) if we follow the Google style guide. I will go ahead and swap the order